### PR TITLE
PERF: 25% reduction in runtime for TreeNode.shear

### PR DESCRIPTION
**Pull request by wasade**
A smarter shearing algorithm. Observed ~25% reduction in runtimes when benchmarking on Greengenes 85% and 88% tree randomly pulling 10% of the tips.

### CODE REVIEW

File: skbio/tree/_tree.py [hunk 1]
```diff
@@ -450,10 +450,19 @@ def shear(self, names):
...
+                marked.add(tip)
+                for anc in tip.ancestors():
+                    if anc in marked:
+                        break
+                    else:
+                        marked.add(anc)
+
+        for node in list(tcopy.traverse()):
+            if node not in marked:
+                node.parent.remove(node)
```
Comment (ebolyen) [inline skbio/tree/_tree.py]: :+1: This uses so fewer iterations it's almost not funny!

**Comment by ebolyen** [comment]
@wasade Looks great, can you add a changelog indicating the performance increase for `shear`?

**Comment by jairideout** [comment]
The 0.5.0-dev changelog entry is missing a `### Performance enhancements` section, can you add it there please?

File: skbio/tree/_tree.py (hunk 1 above)
Comment (wasade) [inline skbio/tree/_tree.py]: Not sure why we didn't just do this simpler method in the first place...

**Comment by jairideout** [comment]
Thanks @wasade!

**Pull request merged**